cmds/run.js

```diff
@@ -22,14 +25,14 @@ exports.handler = (argv) => {
     const jobContents = jobData[1];
     const jobFilePath = jobData[0];
 
-    Promise.resolve()
-        .then(() => Promise.all([tjmFunctions.alreadyRegisteredCheck(jobContents), tjmFunctions.loadAsset()]))
-        .spread((isAlreadyRegistered) => {
-            if (isAlreadyRegistered) {
-                reply.warning(`Job ${jobContents.tjm.job_id} is already registered with cluster ${argv.c}`);
+    tjmFunctions.alreadyRegisteredCheck(jobContents)
+        .catch((err) => {
+            if (err) {
+                reply.warning(err);
             }
-            return tjmFunctions.teraslice.jobs.submit(jobContents)
+            return tjmFunctions.teraslice.jobs.submit(jobContents);
         })
+        .then(() => tjmFunctions.loadAsset())
         .then((jobResult) => {
             const jobId = jobResult.id();
```

Comment:
Seeing the error 'Cannot read property 'id' of undefined' because it's trying to read a job id from the loadAsset function.   Should be reconfigured to register the job if necessary and return the needed job data to either update the job file and run the job or just run the job. Seems like the loadAsset should be at the beginning of the function. (if you are pressed for time with other things, I could finish this up.  Everything else looks really good and is a huge improvement)